Doppler.Sap/Factory/BillingRequestHandler.cs

```diff
@@ -45,60 +45,25 @@ public async Task<SapTaskResult> Handle(SapTask dequeuedTask)
             {
                 var sapSystem = SapSystemHelper.GetSapSystemByBillingSystem(dequeuedTask.BillingRequest.BillingSystemId);
                 var sapTaskHandler = _sapServiceSettingsFactory.CreateHandler(sapSystem);
-                var businessPartner = await sapTaskHandler.TryGetBusinessPartner(dequeuedTask.BillingRequest.UserId, dequeuedTask.BillingRequest.FiscalID, dequeuedTask.BillingRequest.PlanType);
+                var existentInvoice = await sapTaskHandler.TryGeInvoiceByInvoiceId(dequeuedTask.BillingRequest.InvoiceId);
+                var sapResponse = (dequeuedTask.TaskType == Enums.SapTaskEnum.BillingRequest) ? await CreateInvoice(dequeuedTask, sapSystem) : await UpdateInvoice(dequeuedTask, sapSystem, existentInvoice);
```

Comment:
What do you think use the variable existentInvoice :
```suggestion
                var sapResponse = existentInvoice == null ? await CreateInvoice(dequeuedTask, sapSystem) : await UpdateInvoice(dequeuedTask, sapSystem, existentInvoice);
```

Reply:
Let me review it because I don't remember what is reason that I used the enum instead of the variable.

Reply:
@nreal, I used the enum because is possible that in the ```Billing Update``` that the property ```U_DPL_INV_ID == null``` in SAP (we use it property to search in SAP) and in it case is wrong create a new invoice because the invoice was previosly created the correct behavior is try to update the invoice. Let me know if not clear for you and we can have a call.